text/0000-ternary.md

```diff
@@ -0,0 +1,238 @@
++ Feature name: (fill me in with a unique ident, `my_awesome_feature`)
++ Start date: (fill me in with today's date, YYYY-MM-DD)
++ RFC PR: [iotaledger/bee-rfcs#0000](https://github.com/iotaledger/bee-rfcs/pull/0000)
++ Bee issue: [iotaledger/bee#0000](https://github.com/iotaledger/bee/issues/0000)
+
+# Summary
+
+In this RFC we introduce both the API and implementation of the `bee-ternary`
+crate, a general-purpose ternary manipulation crate written in Rust that may be
+used to handle ternary data in the IOTA ecosystem.
+
+# Motivation
+
+Ternary is a fundamental part of the IOTA ecosystem and is used throughout, even
+for fundamental features like wallet addresses and transaction identification.
+
+Manipulating ternary in an ergonomic manner on top of existing binary-driven
+code can require complexity and, up until now, comes either with a lot of
+boilerplate code, implementations that are difficult to verify as correct, or a
+lack of features.
+
+`bee-ternary` seeks to become the canonical ternary crate in the Rust IOTA
+ecosystem by providing an API that is efficient, ergonomic, and featureful all
+at once.
+
+We hope that `bee-ternary` will allow IOTA developers to more easily write code
+that works with ternary and allow the simplification of existing codebases
+through use of the API.
+
+Broadly, there are 3 main benefits things that `bee-ternary` aims to provide
+over use-specific implementations of ternary manipulation code.
+
+- **Ergonomics**: The API should be trivial to use correctly and should
+naturally guide developers towards correct and efficient solutions.
+
+- **Features**: The API should provide a powerful set of core features that can
+be used together to cover most developer requirements. Examples of such features
+include multiple encoding schemes, binary/ternary conversion, (de)serialization,
+and tryte string de/encoding.
+
+- **Performance**: The API should allow 'acceptably' efficient manipulation of
+ternary data. This is difficult to ruggedly define, and implementation details
+may change later, but broadly we intend to introduce an API that doesn't
+inherently inhibit high performance through poor design choices.
+
+# Detailed design
+
+`bee-ternary` is designed to be extensible and is built on top of a handful of
+powerful abstraction types. Below we list the features of the API and summarise
+a few of the core API objects.
```

Comment:
Can you expand on why you talk about *objects* instead of *types*. I you don't want to be too explicit here, I would prefer the term *features*.

Reply:
I use the term to talk about distinct API things, but this could include traits. I'll use the term 'features' if you wish.

Reply:
Not bothered so much tbh :smile: . If you think it's fine, then go for it. My problem with it is that it has different semantics attached to it depending on the context, but essentially it's really just a minor thing. You can close this conversation, if you're sure, that it won't cause confusion (alternatives: features, models, concepts)

Reply:
Have changed it now